gnucash/gnome-utils/gnc-autoclear.cpp

```diff
@@ -250,27 +213,40 @@ gnc_autoclear_get_splits (Account *account, gnc_numeric toclear_value,
 
     status_update (label, _("Cleaning up..."));
 
-    toclear_list = (GList *)g_hash_table_lookup (sack, (gpointer)toclear_value.num);
+    /*
+    for (auto& [thisvalue, splits] : sack)
+    {
+        printf ("dump: %" PRId64 " = ", thisvalue);
+        if (splits == DUP_LIST)
+            printf (" DUPE");
+        else
+            for (GList *n = splits; n; n = n->next)
+                printf (" [%5.2f]", gnc_numeric_to_double (xaccSplitGetAmount ((Split*)n->data)));
+        printf ("\n");
+    }
+    */
 
     /* Check solution */
-    if (!toclear_list)
+    if (sack.find (toclear_value.num) == sack.end())
     {
         g_set_error (error, autoclear_quark (), AUTOCLEAR_UNABLE,
                      _("The selected amount cannot be cleared."));
         goto skip_knapsack;
     }
     else
-        /* copy GList because GHashTable value will be freed */
-        *splits = g_list_copy (toclear_list);
+        /* copy GList because std::unordered_map value will be freed */
+        *splits = g_list_copy (sack[toclear_value.num]);
```

Comment:
In theory if you don't need the value of the std::unordered_map further down you should be able to std::move it into the splits variable. That will avoid an expensive copy. Only theory, not tested yet.